core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize request content before logging. It is common to have the
+     * {@link Function} remove sensitive content, such as an GPS location query, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public T requestContentSanitizer(Function<Object, Object> requestContentSanitizer) {
+        this.requestContentSanitizer = requireNonNull(requestContentSanitizer, "requestContentSanitizer");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link Function} to use to sanitize request content before logging.
+     */
+    public Function<Object, Object> getRequestContentSanitizer() {
+        return requestContentSanitizer;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize response headers before logging. It is common to have the
+     * {@link Function} remove sensitive headers, like {@code Set-Cookie}, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public T responseHeadersSanitizer(Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer) {
+        this.responseHeadersSanitizer = requireNonNull(responseHeadersSanitizer, "responseHeadersSanitizer");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link Function} to use to sanitize response headers before logging.
+     */
+    public Function<HttpHeaders, HttpHeaders> getResponseHeadersSanitizer() {
+        return responseHeadersSanitizer;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize response content before logging. It is common to have the
+     * {@link Function} remove sensitive content, such as an address, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public T responseContentSanitizer(Function<Object, Object> responseContentSanitizer) {
+        this.responseContentSanitizer = requireNonNull(responseContentSanitizer, "responseContentSanitizer");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link Function} to use to sanitize response content before logging.
+     */
+    public Function<Object, Object> getResponseContentSanitizer() {
+        return responseContentSanitizer;
+    }
+
+    /**
+     * Sets the rate at which to sample requests to log. Any number between {@code 0.0} and {@code 1.0} will
+     * cause a random sample of the requests to be logged. The random sampling is appropriate for low-traffic
+     * (ex servers that each receive &lt;100K requests). If unset, all requests will be logged.
+     */
+    public T samplingRate(float samplingRate) {
+        checkArgument(0.0 <= samplingRate && samplingRate <= 1.0, "samplingRate must be between 0.0 and 1.0");
+        this.samplingRate = samplingRate;
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the rate at which to sample requests to log.
+     */
+    public float getSamplingRate() {
```

Comment:
- Remove `get`?
- protected? (a subclass can override it to make it accessible from the classes from the same package)